api/catalog/api/utils/validate_images.py

```diff
@@ -10,6 +10,14 @@
 parent_logger = logging.getLogger(__name__)
 
 
+CACHE_PREFIX = "valid:"
+
+
+def _get_cached_statuses(redis, image_urls):
+    cached_statuses = redis.mget([CACHE_PREFIX + url for url in image_urls])
+    return [int(b.decode("utf-8")) if b is not None else None for b in cached_statuses]
```

Comment:
This change is to facilitate mocking the cached statuses in the test. Mostly this is to avoid the status cache being prepopulated based on other tests request/response cycle.

Reply:
It's also welcome modularization!